src/amo/components/CollectionList/index.js

```diff
@@ -0,0 +1,174 @@
+  location: ReactRouterLocation,
+|};
+
+export class CollectionListBase extends React.Component<Props> {
+  static defaultProps = {
+    creating: false,
+  };
+
+  componentDidMount() {
+    const {
+      collections,
+      currentUsername,
+      dispatch,
+      errorHandler,
+      loadingUserCollections,
+    } = this.props;
+
+    if (currentUsername && !loadingUserCollections && !collections) {
+      dispatch(fetchUserCollections({
+        errorHandlerId: errorHandler.id, username: currentUsername,
+      }));
+    }
+  }
+
+  render() {
+    const { i18n, isLoggedIn, location, collections } = this.props;
+    const myCollectionsHeader = i18n.gettext('My Collections');
+
+    if (!isLoggedIn) {
+      return (
+        <AuthenticateButton
+          noIcon
+          location={location}
+          logInText={i18n.gettext('Log in to view your collections')}
+        />
+      );
+    }
+
+    const collectionElements = [];
+
+    if (collections && collections.length) {
+      collections.forEach((collection) => {
+        const { authorUsername, id, name, numberOfAddons, slug } = collection;
+        const numberText = i18n.sprintf(
+          i18n.ngettext(
+            '%(total)s add-on', '%(total)s add-ons', numberOfAddons
+          ),
+          { total: i18n.formatNumber(numberOfAddons) },
+        );
+        collectionElements.push(
+          <li className="CollectionList-collection" key={id}>
+            <a
+              className="CollectionList-collection-link"
+              href={`/collections/${authorUsername}/${slug}/`}
+            >
+              <h2 className="CollectionList-collection-name">
+                {name}
+              </h2>
+              <p className="CollectionList-collection-number">
+                {numberText}
+              </p>
+            </a>
+          </li>
+        );
+      });
+    }
+
+    return (
+      <div className="CollectionList">
+        <div className="CollectionList-wrapper">
+          <Card
+            className="CollectionList-info"
+            header={myCollectionsHeader}
+            key="info"
+          >
+            <p className="CollectionList-info-text">
+              {i18n.gettext(`Collections make it easy to keep track of favorite
+                add-ons and share your perfectly customized browser with others.`)}
+            </p>
+            <Button
+              buttonType="action"
+              className="CollectionList-create"
+              puffy
+              to="/collections/add/"
+              type="button"
+            >
+              {i18n.gettext('Create a collection')}
+            </Button>
+          </Card>
+          <Card
+            className="CollectionList-list"
+            header={myCollectionsHeader}
+            key="list"
+          >
+            {collectionElements.length ? (
+              <ul className="CollectionList-listing">
+                {collectionElements}
+              </ul>
+            ) : null}
```

Comment:
Hum, we should probably have some `LoadingText` because it looks a bit broken when waiting for the data to be retrieved:

![screen shot 2018-06-22 at 14 10 52](https://user-images.githubusercontent.com/217628/41776109-bdc720ba-7626-11e8-9760-981f1b2f8fe0.png)

Reply:
I also noticed that we don't have a special "empty state" when the user does not have any collections yet. It shows the same page as above ^